### PR TITLE
feat: add btc app configs for ltc and doge

### PR DESCRIPTION
**Pull request by natto1784**
Implements https://app.clickup.com/t/9002019994/CHI-5062
Implements https://app.clickup.com/t/9002019994/CHI-5063

### CODE REVIEW

File: flake.nix
Comment (ujjwal-cyph) [inline flake.nix:1]: added by mistake?

File: utilities/build.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
```
Comment (ujjwal-cyph) [inline utilities/build.sh:1]: let's take this as a separate pull request

File: apps/btc_family/ltc/ltc_app.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+static bool is_purpose_supported(uint32_t purpose_index);
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/**
+ * Litecoin supports Segwit and Native-Segwit but not Taproot unlike Bitcoin
+ * Values can be verified from the following references:
+ * https://github.com/litecoin-project/litecoin
```
Comment (ujjwal-cyph) [inline apps/btc_family/ltc/ltc_app.c:101]: i couldn't find anything here. is this reference useful?

File: apps/btc_family/doge/doge_app.c [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/**
+ * Dogecoin only supports legacy addresses. No support for Segwit and
+ * Native-Segwit. Hence, bech32_hrp is empty and segwit, native-segwit xpub
+ * version is set to 0.
+ * Secondary reference for aggregated coin info:
+ * https://github.com/trezor/trezor-firmware/blob/f5983e7843f381423f30b8bc2ffc46e496775e5a/common/defs/bitcoin/doge.json
```
Comment (ujjwal-cyph) [inline apps/btc_family/doge/doge_app.c:103]: incorrect reference

File: apps/btc_family/doge/doge_app.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief Returns the config for Dogecoin chain app
+ *
+ * @return A const reference to doge_config_t
```
Comment (ujjwal-cyph) [inline apps/btc_family/doge/doge_app.h:39]: `doge_config_t` is not a valid struct

correct at other place too.

File: flake.nix
Comment (natto1784) [reply flake.nix:1]: seems so

File: apps/btc_family/doge/doge_app.c [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+ */
+const btc_config_t doge_app = {
+    .coin_type = COIN_DOGE,
+    .p2pkh_addr_ver = 0x1e,
+    .p2sh_addr_ver = 0x20,
```
Comment (ujjwal-cyph) [inline apps/btc_family/doge/doge_app.c:108]: wrong value. should be `22` in decimal

File: apps/btc_family/ltc/ltc_app.c (hunk 2 above)
Comment (natto1784) [reply apps/btc_family/ltc/ltc_app.c:101]: [the chainparams](https://github.com/litecoin-project/litecoin/blob/master/src/chainparams.cpp) have some of the info

Comment (ujjwal-cyph) [reply apps/btc_family/ltc/ltc_app.c:101]: oh!

in that case, [use this link](https://github.com/litecoin-project/litecoin/blob/5ac781487cc9589131437b23c69829f04002b97e/src/chainparams.cpp#L65-L177)

Comment (natto1784) [reply apps/btc_family/ltc/ltc_app.c]: done

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-5063 Implement Dogecoin](https://app.clickup.com/t/9002019994/CHI-5063)

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-5062 Implement Litecoin](https://app.clickup.com/t/9002019994/CHI-5062)

**Pull request merged**